src/runtime_src/core/tools/xbmgmt2/OO_LoadConfig.cpp

```diff
@@ -126,5 +128,44 @@ OO_LoadConfig::execute(const SubCmdOptions& _options) const
     return;
   }
 
-  //TO_DO: parse the INI file
-}
\ No newline at end of file
+  try {
+    for (auto& t : deviceCollection)  
+      loadconfig(t);
+  } catch (const std::runtime_error& e) {
+    // Catch only the exceptions that we have generated earlier
+    std::cerr << boost::format("ERROR: %s\n") % e.what();
+    return;
+  }
+}
+
+/*
+ * so far, we only support the following configs
+ * [Device]
+ * mailbox_channel_disable = xxx
+ * mailbox_channel_switch = xxx
+ * cahce_xclbin = xxx
+ */
+void
+OO_LoadConfig::loadconfig(const std::shared_ptr<xrt_core::device>& _dev) const
+{
+  boost::property_tree::ptree m_tree;
+  boost::property_tree::ini_parser::read_ini(m_path, m_tree);
+
+  for (auto& it : m_tree) {

```

Comment:
This nesting of boost property tree doesn't not clearly indicate the intent and assumptions are being made here regarding the presents of values.

Please update this algorithm to:

1. Use boost to retrieve the "Devices" child tree instead of doing a linear search.  For example:
`const boost::property_tree::ptree ptDevices = m_tree.get_child("Devices", empty_ptree<ptree>()); `

2. Validate that there are entries in the ptDevices property tree.
3. Request, via boost each entry of interest, be sure to include a "default value" in case it isn't present.

Reply:
will change.
BTW, default is not required since the driver already has default for each of the config items already. so if not present, just means the user doesn't try to config with non-default value